analyzer/main.zeek

```diff
@@ -50,9 +65,137 @@ export {
+{
+	if ( ! pe_log_section_flags ) {
+		return;
+	}
 
+    # The string that holds the one-character flags, "r", "w" and "e"
+    local flag_string: string = "";
+
+    # Only iterate over the chars once and check if some flags are set
+    # IMAGE_SCN_MEM_EXECUTE     = 0x20000000 -> The section can be executed as code.
+    # IMAGE_SCN_MEM_READ        = 0x40000000 -> The section can be read.
+    # IMAGE_SCN_MEM_WRITE       = 0x80000000 -> The section can be written to.
+
+    # https://docs.microsoft.com/en-us/windows/win32/debug/pe-format#section-flags
+    for ( flag in h$characteristics ) {
+        if ( flag == 0x40000000 ) {
+            flag_string += "r";
+        }
+        if ( flag == 0x80000000 ) {
+            flag_string += "w";
+        }
+        if ( flag == 0x20000000 ) {
+            flag_string += "e";
+        }
+    }
+
+	if ( h$name !in f$pe$section_info_table ) {
+		f$pe$section_info_table[h$name] = [];
+	}
+	f$pe$section_info_table[h$name]$flags = flag_string;
+
+}
+
+event pe_import_table(f: fa_file, it: PE::ImportTable) {
+	if ( ! pe_log_import_table ) {
+		return;
+	}
+    # The vector that we're going to fill
+    local temp_tbl:  vector of string;
+
+    # Iterate over the import table entries
+    for ( i in it$entries ) {
+
+        local e = it$entries[i];
+
+        # If there are any imports....
+        if ( e?$imports ) {
+
+            # ... iterate over them
+            for ( j in e$imports ) {
+
+                # And for every imported function, check whether it's imported by name or ordinal
+                # Add the corresponding information to the vector of strings
+                local imp = e$imports[j];
+
+                if ( imp?$hint_name_rva ) {
+                    temp_tbl += fmt("%s:%s", e?$dll ? e$dll : "nil", imp?$name ? imp$name : "<nil>");
+                }
+                else {
+                    temp_tbl += fmt("%s:%s", e?$dll ? e$dll : "nil", imp$ordinal);
+                }
+            }
+        }
+    }
+    # Finally, put it in the actual PE log
+    f$pe$import_table = temp_tbl;
 }
 
+# Called when the file analysis is closed
+event file_state_remove(f: fa_file)
+    {
+	# If any of the new logging is enabled, delete the default section_names field
+	# This means that default functionality is not changed
+	if ( pe_log_section_flags  || pe_log_section_entropy ) {
+		f$pe$section_names = vector();
+
+		for ( section, info in f$pe$section_info_table ) {
+			local formatted_string: string = "";
+			if (pe_log_section_entropy && pe_log_section_flags ) {
+				formatted_string = fmt("%s:%s:%.2f", section, info$flags, info$entropy);
+			}
+			if (pe_log_section_entropy && ! pe_log_section_flags ) {
+				formatted_string = fmt("%s:%.2f", section, info$entropy);
+			}
+			if (! pe_log_section_entropy && pe_log_section_flags ) {
+				formatted_string = fmt("%s:%s", section, info$flags);
+			}
+			f$pe$section_names += formatted_string;
+		}
+	}
+}
```

Comment:
The different conditionals inside the loop are getting hard to follow, how about the following instead?
```suggestion

# Called when the file analysis is closed
event file_state_remove(f: fa_file)
    {
	# If any of the detailed logging is enabled, delete the default section_names field.
	# This means that default functionality is not changed.
	if ( pe_log_section_flags  || pe_log_section_entropy )
		{
		f$pe$section_names = vector();

		for ( section, info in f$pe$section_info_table )
			{
			local log_entropy = pe_log_section_entropy ? fmt("%.2f", info$entropy) : "";
			local log_section_flags = pe_log_section_flags ? fmt("%s", info$flags) : "";
			local log_frag = vector(log_section_flags, log_entropy);

			local formatted_string: string = fmt("%s", section);
			for (i in log_frag)
				{
				if (|log_frag[i]| != 0)
					{
					formatted_string += fmt(":%s", log_frag[i]);
					}
				}

			f$pe$section_names += formatted_string;
			}
		}
	}
```

Reply:
Agree, I committed your suggestion.